Sources/ApodiniTypeDescriptor/InstanceCreator.swift

```diff
@@ -0,0 +1,122 @@
+                throw InstanceCreatorError.nonSupportedDictionaryKey(keyType)
+            }
+            instance = try InstanceCreator.dictionaryInstance(key: primitiveType, valueType: valueType)
+        } else if mangledName == .optional,
+                  let wrappedValueType = genericTypes.first {
+            instance = try InstanceCreator(type: wrappedValueType).instance
+        } else {
+            instance = try createInstance(of: type)
+            try handleEmptyProperties()
+        }
+    }
+    
+    mutating func handleEmptyProperties() throws {
+        let typeInfo = try Runtime.typeInfo(of: type(of: instance))
+        
+        try typeInfo.properties
+            .filter { $0.type is Encodable.Type }
+            .forEach { property in
+                try handleOptional(for: property)
+                try handleDictionary(for: property)
+                try handleArray(for: property)
+            }
+    }
+    
+    mutating func handleOptional(for property: PropertyInfo) throws {
+        guard
+            MangledName(property.type) == .optional,
+            let wrappedValueType = try? Runtime.typeInfo(of: property.type).genericTypes.first
+        else { return }
+        
+        try property.set(value: try InstanceCreator(type: wrappedValueType).instance, on: &instance)
+    }
+    
+    mutating func handleDictionary(for property: PropertyInfo) throws {
+        guard MangledName(property.type) == .dictionary else {
+            return
+        }
+        
+        let genericTypes = try Runtime.typeInfo(of: property.type).genericTypes
+        guard
+            let keyType = genericTypes.first,
+            let valueType = genericTypes.last
+        else { return }
+        
+        guard let primitiveType = PrimitiveType(keyType) else {
+            throw InstanceCreatorError.nonSupportedDictionaryKey(keyType)
+        }
+        
+        let propertyInstance = try InstanceCreator.dictionaryInstance(key: primitiveType, valueType: valueType)
+        try property.set(value: propertyInstance, on: &instance)
+    }
+    
+    mutating func handleArray(for property: PropertyInfo) throws {
+        guard
+            MangledName(property.type) == .array,
+            let elementType = try? Runtime.typeInfo(of: property.type).genericTypes.first
+        else { return }
+        
+        try property.set(value: [try InstanceCreator(type: elementType).instance], on: &instance)
+    }
+    
+    func typedInstance<T: Encodable>(_ type: T.Type) throws -> T {
+        guard let instance = instance as? T else {
+            throw InstanceCreatorError.failedCastingInstanceToEncodable
+        }
+        return instance
+    }
+}
+
+extension InstanceCreator {
+    static func dictionaryInstance(key: PrimitiveType, valueType: Any.Type) throws -> Any {
+        let valueInstance = try InstanceCreator(type: valueType).instance
+        switch key {
+        case .int: return [Int(): valueInstance]
```

Comment:
Just testing for primitive types here seems really restraining. I haven't tested it, but wouldn't it be possible to do something like this?

```swift
static func dictionaryInstance(keyType: Any.Type, valueType: Any.Type) throws -> Any {
    let valueInstance = try InstanceCreator(type: valueType).instance
    let keyInstanceCandidate = try InstanceCreator(type: keyType).instance
    guard let keyInstance = keyInstanceCandidate as? Hashable else {
        throw InstanceCreatorError.nonSupportedDictionaryKey(keyType)
    }
    return [keyInstance: valueInstance]
}
```